src/libslic3r/GCode.cpp

```diff
@@ -1918,6 +1875,15 @@ void GCode::_do_export(Print& print, GCodeOutputStream &file, ThumbnailsGenerato
     } else
 	    m_enable_extrusion_role_markers = false;
 
+    // if thumbnail type of BIQU, insert above header
+    // if not, it is inserted under the header in its normal spot
+    const GCodeThumbnailsFormat m_gcode_thumbnail_format = print.full_print_config().opt_enum<GCodeThumbnailsFormat>("thumbnails_format");
```

Comment:
Hi @Ocraftyone, quick question: Which specific device/machine is this format intended to support? Is it just for a single printer model from BIQU? More importantly, is the position change absolutely necessary? Perhaps we can encourage them to make their machine compatible with the de facto standards used in all PrusaSlicer derivatives?

I'm just being cautious here. I'm facing increasing maintenance and testing challenges. A major reason for this is the numerous special case handling codes in the current codebase. @_@

Reply:
No problem! This is for the BigTreeTech TFTs. This implementation is heavily based on the first party cura plugin. The position with the current implementation is necessary. I did see references to a way to not require the thumbnail at the top of the file, but I was just trying to get it to work and this is how the cura plugin does it. There is also an option that allows the display to decide a base64 png, but this requires firmware modification to enable the flag and is much slower than the native format as all the conversion is done display side. Links and more info on the implementation are in the linked issue. All testing and results were done there.

Reply:
Noted.
Let's keep it then.

BTW, would it be better to name it as BTT_TFT instead of BIQU?

Reply:
Sure. I am absolutely open to that! I was just following the format that SuperSlicer followed (SuperMerill made an attempt to get this working, but did not complete it. Helped with some development though!).